thorlcr/graph/thgraphslave.cpp

```diff
@@ -140,8 +140,29 @@ void CSlaveActivity::setInput(unsigned index, CActivityBase *inputActivity, unsi
     else
         outLink.set(((CSlaveActivity *)inputActivity)->queryOutput(inputOutIdx));
     assertex(outLink);
+
     while (inputs.ordinality()<=index) inputs.append(NULL);
-    inputs.replace(outLink.getClear(), index);
+
+    inputs.replace( outLink.getClear(), index);
+}
+
+void CSlaveActivity::appendOutput(IThorDataLink *itdl)
+{
+    if (queryJob().getOptBool("TRACEROWS"))
+    {
+        const unsigned numTraceRows = queryJob().getOptInt("numTraceRows", 10);
+        Owned<IThorDataLink> newItdl;
+        newItdl.setown(new CTracingThorDataLink(itdl,queryHelper(),numTraceRows));
```

Comment:
probably no point here in the newItdl declaration, setown, then getClear(), i.e. may as well do:
```
outputs.append(new CTracingThorDataLink(itdl,queryHelper(),numTraceRows));
```